src/transport/advection_diffusion_model.hh

```diff
@@ -85,16 +93,80 @@ public:
 			std::vector< arma::vec > &init_values) = 0;
 
 	/**
+	 * Return types of boundary conditions for each solution component.
+	 * @param ele_acc  Element accessor.
+	 * @param bc_types Vector of bc. types (output, see BC_Type)
+	 */
+	virtual void get_bc_type(const ElementAccessor<3> &ele_acc,
+			arma::uvec &bc_types) = 0;
+
+	/**
 	 * Computes the Dirichlet boundary condition values.
 	 * @param point_list   Points at which to evaluate.
 	 * @param ele_acc      Element accessor.
 	 * @param bc_values    Vector of b.c. values (output).
 	 */
-	virtual void compute_dirichlet_bc(const std::vector<arma::vec3> &point_list,
+	virtual void get_dirichlet_bc_data(const std::vector<arma::vec3> &point_list,
```

Comment:
Virtual methods are just for declaration of the common interface. TransportDG does not use virtual calls because it accesses directly the methods from Model.